pkg/apis/pipeline/v1alpha1/pipeline_conversion_test.go

```diff
@@ -27,9 +27,32 @@ import (
 	"github.com/tektoncd/pipeline/test/diff"
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/selection"
 	"knative.dev/pkg/apis"
 )
 
+var ignoreAnnotations = cmp.FilterPath(func(p cmp.Path) bool {
+	if len(p) < 2 {
+		return false
+	}
+	structField, ok := p.Index(-2).(cmp.StructField)
+	if !ok {
+		return false
+	}
+	if structField.Name() != "ObjectMeta" {
+		return false
+	}
+
+	structField, ok = p.Index(-1).(cmp.StructField)
```

Comment:
the -1 and -2 to me feel like they could use a comment or variable to add some more context